org.eclipse.gef/src/org/eclipse/gef/ui/parts/AbstractEditPartViewer.java

```diff
@@ -363,16 +365,17 @@ public Map getEditPartRegistry() {
 	 */
 	@Override
 	public EditPart getFocusEditPart() {
-		if (focusPart != null)
+		if (focusPart != null) {
 			return focusPart;
+		}
 		if (getSelectedEditParts().isEmpty()) {
-			if (getContents() != null)
+			if (getContents() != null) {
 				return getContents();
-			else
-				return getRootEditPart();
+			}
+			return getRootEditPart();
 		}
-		List selection = getSelectedEditParts();
-		return (EditPart) selection.get(selection.size() - 1);
+		List<? extends EditPart> selection = getSelectedEditParts();
+		return selection.get(selection.size() - 1);
```

Comment:
I can't wait for Java 21 where we can simply call selection.getLast() :sweat_smile:

Reply:
me too